### ISSUE TITLE
StringIndexOutOfBoundsException while processing documents

### ISSUE THREAD

**Issue by schoch** [tabulapdf/tabula-java#37]
Today, I had a couple of StringIndexOutOfBoundsExceptions in `ObjectExtractor.isPrintable(...)`. The reason for that was/is a String with length 0. I spend some time with my debugger, but was not able to finally figure out if that is a bug in tabula or in my pdf documents.

So, before I go deeper inside the code... does anybody have any ideas about that topic? If it is just a bug in the `isPrintable` check, it would be easy to fix.

**Comment by jeremybmerrill** [comment]
Manuel's the expert, but it looks to me like an easy-enough-to-fix bug in ObjectExtractor.java: https://github.com/tabulapdf/tabula-java/blob/master/src/main/java/technology/tabula/ObjectExtractor.java#L426-L431

**Comment by jazzido** [comment]
`isPrintable` should not receive a 0-length string as the argument, that should probably be catched on or before [`ObjectExtractor.processTextPosition`](https://github.com/tabulapdf/tabula-java/blob/master/src/main/java/technology/tabula/ObjectExtractor.java#L322 <https://github.com/tabulapdf/tabula-java/blob/master/src/main/java/technology/tabula/ObjectExtractor.java#L322>). 

Can you reproduce the bug and write a failing test case?

**Comment by jazzido** [comment]
`isPrintable` should not receive a 0-length string as the argument, that should probably be catched on or before [`ObjectExtractor.processTextPosition`](https://github.com/tabulapdf/tabula-java/blob/master/src/main/java/technology/tabula/ObjectExtractor.java#L322 <https://github.com/tabulapdf/tabula-java/blob/master/src/main/java/technology/tabula/ObjectExtractor.java#L322). 

Can you reproduce the bug and write a failing test case?

**Comment by schoch** [comment]
Yes, I can do that. Hope that I find the time in the next few days.

**Comment by jazzido** [comment]
Thanks!
